core/src/io/github/lonamiwebs/klooni/game/State.java

```diff
@@ -0,0 +1,282 @@
+package io.github.lonamiwebs.klooni.game;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+
+/*
+    State is an object designed to help validating randomly-generated blocks. Every time a block is
+    inserted into the board, the game's state changes. Different permutation of blocks and different
+    position creates new states. Thus to validate a set of generated blocks, we have to check all
+    possible states. Instead of checking them directly on the board, we create a special class for
+    validation.
+
+    This is used in infinite mode.
+ */
+
+public class State {
```

Comment:
So if I understood correctly, `State` is pretty much `Board`, but without worrying about the UI. That makes me wonder if we should have a `State` and let `Board` inherit it, or relate them somehow. So you can run simulations without the UI, and not needing to copy-paste so much code (all `inBounds`, `putPiece`, `clearBoard` etc.)

Reply:
Should I do that? For now, I can't think of a way for `Board` to inherit `State`, since their similarities are in the functions. I should consult this first 😄

Reply:
I don't think it's necessary for now. This can always be improved later.